sdk/core/Azure.Core.TestFramework/src/TestEnvironment.cs

```diff
@@ -243,8 +243,18 @@ public virtual TokenCredential Credential
                     }
                     else
                     {
-                        _credential = new DefaultAzureCredential(
-                             new DefaultAzureCredentialOptions { ExcludeManagedIdentityCredential = true });
+                        var defaultAzureCredentialOptions = new DefaultAzureCredentialOptions
+                        {
+                            ExcludeManagedIdentityCredential = true
+                        };
+
+                        var tenantId = GetVariable("TENANT_ID");
+                        if (!string.IsNullOrWhiteSpace(tenantId))
```

Comment:
@christothes:  Should we be setting any kind of additionally allowed tenants here to ensure the widest compatibility for local + CI runs?

Reply:
AdditionallyAllowedTenants only applies for certain services that support tenant discovery. Setting the AZURE_TENANT_ID env variable would be the correct way to specify a tenant other than the identity's home tenant.

Reply:
@christothes: Thanks!   Does that imply that we don't need the changes to test environment here and `AzZURE_TENANT_ID` would get picked up automatically - or that L251 should have the name adjusted and continue to set the tenant explicitly in options?

Reply:
If we use DefaultAzureCredential, it would be automatic. It's true for some other credentials also as the default value if options are not provided with a TenantId specified.